BetterPasswordPlugin.inc.php

```diff
@@ -158,25 +158,15 @@ function checkPassword($hookName, $args) {
 			return false;
 		}
 		if ($this->getSetting(CONTEXT_SITE, 'betterPasswordCheckBlacklist')) {
-			$badPassword = false;
 			$lowerPassword = strtolower($password);
-			foreach ($this->getBlacklists() as $filename) {
-				/* In case of out-of-memory error, break glass
-				$passwordfile = fopen($filename, "r");
-				while (!feof($passwordfile)) {
-					$disallowed = fgets($passwordfile);
-					if ($lowerPassword === $disallowed) {
-						$badPassword = true;
-						break;
-					}
-				}
-				fclose($passwordfile);
-				 */
-				$passwordfile = file_get_contents($filename);
-				if (strpos($passwordfile, $lowerPassword) !== false) {
-					$badPassword = true;
-					break;
-				}
+			$shaPass = sha1($lowerPassword);
+			$passwordHash = substr($shaPass,0,2);
+			$blacklist = $this->generateBlacklist();
+			if ($blacklist) {
+				$cache = CacheManager::getManager()->getCache('badPasswords', $passwordHash, array($this, '_PasswordCacheMiss'));
+				$badPassword = $cache->get($shaPass);
+			} else {
+				$form->addError($errorField, __('plugins.generic.betterPassword.validation.betterPasswordUnexpectedError'));
```

Comment:
This error message was actually included in the uploadBlacklists PR, rather than this one.